dask/dataframe/core.py

```diff
@@ -1451,13 +1451,29 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
                                     axis=axis, skipna=skipna, ddof=ddof)
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            num = self.select_dtypes(include=['number', np.timedelta64, 'bool']) \
+                if is_dataframe_like(self) else self
```

Comment:
Maybe `if self.ndim > 1`? I had to pause for a second to think about why `is_dataframe_like(self)` was needed.